pkg/cmd/api/api.go

```diff
@@ -36,15 +37,17 @@ type ApiOptions struct {
 	Paginate            bool
 	Silent              bool
 
-	HttpClient func() (*http.Client, error)
-	BaseRepo   func() (ghrepo.Interface, error)
+	HttpClient    func() (*http.Client, error)
+	BaseRepo      func() (ghrepo.Interface, error)
+	CurrentBranch func() (string, error)
 }
 
 func NewCmdApi(f *cmdutil.Factory, runF func(*ApiOptions) error) *cobra.Command {
 	opts := ApiOptions{
-		IO:         f.IOStreams,
-		HttpClient: f.HttpClient,
-		BaseRepo:   f.BaseRepo,
+		IO:            f.IOStreams,
+		HttpClient:    f.HttpClient,
+		BaseRepo:      f.BaseRepo,
+		CurrentBranch: git.CurrentBranch,
```

Comment:
Instead of `git.CurrentBranch`, you can pass `f.Branch`, since this is already what a `cmdutil.Factory` provides and allows the current branch to be mocked easier in tests